File: 2502-sort-the-people/sort-the-people.java
class Solution {
    static class Mycomp implements Comparator<Integer>{
        int[] heights;
        Mycomp(int[] heights){
            this.heights=heights;
        }
        public int compare(Integer i1,Integer i2){
            return (int)Math.signum((double)(heights[i2]-heights[i1]));
        }
    };
    public String[] sortPeople(String[] names, int[] heights) {
        int n=names.length;
        String[] ans=new String[n];
        Integer[] indices=new Integer[n];
        for(int i=0;i<n;i++)
            indices[i]=i;
        
        Arrays.sort(indices,new Mycomp(heights));
        for(int i=0;i<n;i++)
            ans[i]=names[indices[i]];
        
        return ans;
    }
}

File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<int> ind(names.size());
        iota(ind.begin(),ind.end(),0);
        sort(ind.begin(),ind.end(),[&heights](int i1,int i2){
            if(heights[i1]==heights[i2]) return false;
            return heights[i1]>heights[i2];
        });
        vector<string> ans;
        for(int i=0;i<names.size();i++)
            ans.push_back(names[ind[i]]);
        return ans;
    }
};